src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
That is the crux. We need a way to provide the appropriate core name from inside the `SolrTemplate`. In that case we don't need any core name inside `SolrOperations`. The implementer would add it to the executor call.

Or maybe we can 'find' `SolrOperations` for a core name and provide an appropriately prepared instance. Just because solrCore is already an attribute in SolrTemplate.